lib/job/arFileImportJob.class.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Job worker for XML and CSV imports initiated from the WebUI.
+ *
+ * @package    symfony
+ * @subpackage jobs
+ */
+
+class arFileImportJob extends arBaseJob
+{
+  /**
+   * @see arBaseJob::$requiredParameters
+   */
+  public function runJob($parameters)
+  {
+    $this->info($this->i18n->__('Importing %1 file: %2.', array('%1' => strtoupper($parameters['importType']), '%2' => $parameters['file']['name'])));
+
+    // Set indexing preference.
+    if (false == $parameters['index'])
+    {
+      QubitSearch::disable();
+    }
+
+    try
+    {
+      switch ($parameters['importType'])
+      {
+        case 'csv':
+          $importer = new QubitCsvImport;
+
+          $this->setCsvImportParams($importer, $parameters);
+
+          $importer->import($parameters['file']['tmp_name'], $parameters['objectType'], $parameters['file']['name']);
+
+          break;
+
+        case 'xml':
+          $importer = new QubitXmlImport;
+
+          $options = $this->setXmlImportParams($importer, $parameters);
+
+          $importer->import($parameters['file']['tmp_name'], $options, $parameters['file']['name']);
+
+          break;
+
+        default:
+          // 'importType' defaults to 'CSV' by design if extension is blank or something unknown.
+          // This was to prevent errors if csv file does not have the correct extension. See
+          // modules/object/actions/importAction.class.php.  This default case should never be called.
+          $this->error($this->i18n->__('Unable to import selected file: unknown file extension.'));
+          return false;
+
+          break;
+      }
+    }
+    catch (sfException $e)
+    {
+      $this->error($e->getMessage());
+      return false;
+    }
+
+    if ($importer->hasErrors())
+    {
+      foreach ($importer->getErrors() as $error)
+      {
+        $this->info($error);
+      }
+    }
+
+    // Mark job as complete.
+    $this->info($this->i18n->__('Import complete.'));
+    $this->job->setStatusCompleted();
+    $this->job->save();
+
+    return true;
+  }
+
+  private function setCsvImportParams(&$importer, $parameters)
```

Comment:
it's usually good for non-trivial functions to have a comment block at the top summarizing parameters/return values and the function's purpose. You can see examples elsewhere in the codebase on how we do it if you're not certain